### PR TITLE
persistent transactions fixes

### PR DESCRIPTION
**Pull request by dev-matan-tsuberi**
Some bug fixes

### CODE REVIEW

**Comment by dev-matan-tsuberi** [comment]
[When a user rejects a transaction in Metamask, instead of showing an "Operation failed" message show a "You have cancelled [name of action] on [proposal title]" notification.](https://trello.com/c/uoquRgmD/315-when-a-user-rejects-a-transaction-in-metamask-instead-of-showing-an-operation-failed-message-show-a-you-have-cancelled-name-of-a)

**Comment by dev-matan-tsuberi** [comment]
[Show tx Errors to the user.](https://trello.com/c/KSGzSFuG/238-show-tx-errors-to-the-user)

**Comment by dev-matan-tsuberi** [comment]
[Add an EtherScan link to each pending tx in notifications.](https://trello.com/c/wC7sxrmk/301-add-an-etherscan-link-to-each-pending-tx-in-notifications)

**Comment by dev-matan-tsuberi** [comment]
[Persist unconfirmed transactions for each user](https://trello.com/c/cAxSwDTR/233-persist-unconfirmed-transactions-for-each-user)

**Comment by dev-matan-tsuberi** [comment]
[Users can't read error messages](https://trello.com/c/uX9I91sx/350-users-cant-read-error-messages)

File: src/actions/arcActions.ts [hunk 1]
```diff
@@ -619,6 +619,7 @@ export function createProposal(daoAvatarAddress: string, title: string, descript
       dispatch(push("/dao/" + daoAvatarAddress));
 
       await contributionRewardInstance.proposeContributionReward({
+        title,
```
Comment (tibetsprague) [inline src/actions/arcActions.ts]: huh? why is this here? isn't this calling arc.js, why would you pass in a title?

File: src/reducers/operations.ts [hunk 2]
```diff
@@ -100,6 +101,11 @@ export const operationsTracker: Middleware =
       // discard the `txEventContext` property since it's not serializable and irelevent.
       const {txEventContext: _,  ...options} = info.options;
 
+      let proposalTitle = options.title;
+      if (options.proposalId) {
+        proposalTitle = (getState() as any as IRootState).arc.proposals[options.proposalId].title;
```
Comment (tibetsprague) [inline src/reducers/operations.ts]: as any as IRootState?

File: src/reducers/notifications.ts [hunk 3]
```diff
@@ -176,10 +188,26 @@ const messages: {[key: string]: (state: IRootState, options: any) => string} = {
...
+            'you canceled the tx' :
+          error === OperationError.Reverted ?
+            'the tx was reverted' :
+          error === OperationError.OutOfGas ?
+            'of insufficient gas' :
+            `of unknown reason, contact us at https://discord.gg/3d5C2y7`
+        ) : '';
+
+      let fullErrorMessage;
+      if (error && error !== OperationError.Canceled && error !== OperationError.Canceled && error !== OperationError.Canceled) {
```
Comment (tibetsprague) [inline src/reducers/notifications.ts]: uh, you used the same condition three times

Comment (dev-matan-tsuberi) [reply src/reducers/notifications.ts]: Thanks!
https://github.com/daostack/alchemy/pull/119/commits/771462e487ce5a8046e35910f854c57c9fd61858

File: src/reducers/operations.ts (hunk 2 above)
Comment (dev-matan-tsuberi) [reply src/reducers/operations.ts]: To make Typescript happy since it infers it to be a generic `S`. `getState: <S>() -> S`.

File: src/actions/arcActions.ts (hunk 1 above)
Comment (tibetsprague) [reply src/actions/arcActions.ts]: weird, ok
